sdk/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleMeasure.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import io.opentelemetry.metrics.DoubleMeasure;
+import io.opentelemetry.metrics.DoubleMeasure.BoundDoubleMeasure;
+import io.opentelemetry.metrics.LabelSet;
+import java.util.List;
+import java.util.Map;
+
+class SdkDoubleMeasure extends BaseInstrument<BoundDoubleMeasure> implements DoubleMeasure {
+
+  private final boolean absolute;
+
+  private SdkDoubleMeasure(
+      String name,
+      String description,
+      Map<String, String> constantLabels,
+      List<String> labelKeys,
+      boolean absolute) {
+    super(name, description, constantLabels, labelKeys);
+    this.absolute = absolute;
+  }
+
+  @Override
+  public void record(double value, LabelSet labelSet) {
+    createBoundInstrument(labelSet).record(value);
+  }
+
+  @Override
+  BoundDoubleMeasure createBoundInstrument(LabelSet labelSet) {
+    return new SdkBoundDoubleMeasure(labelSet, this.absolute);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof SdkDoubleMeasure)) {
+      return false;
+    }
+    if (!super.equals(o)) {
+      return false;
+    }
+
+    SdkDoubleMeasure that = (SdkDoubleMeasure) o;
+
+    return absolute == that.absolute;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = super.hashCode();
+    result = 31 * result + (absolute ? 1 : 0);
+    return result;
+  }
+
+  private static class SdkBoundDoubleMeasure extends BaseBoundInstrument
+      implements BoundDoubleMeasure {
+
+    private final boolean absolute;
+
+    SdkBoundDoubleMeasure(LabelSet labels, boolean absolute) {
+      super(labels);
+      this.absolute = absolute;
+    }
+
+    @Override
+    public void record(double value) {
+      if (this.absolute && value < 0) {
+        throw new IllegalArgumentException("absolute measure can only increase");
```

Comment:
increase isn't the right word here. I would say "absolute measures can only record positive values"